n3fit/src/evolven3fit/evolve.py

```diff
@@ -0,0 +1,204 @@
+        dump_evolved_replica(
+            evolved_block, usr_path, int(replica.removeprefix("replica_"))
+        )
+    # remove folder:
+    # The function dump_evolved_replica dumps the replica files in a temporary folder
+    # We need then to remove it after fixing the position of those replica files
+    (usr_path / "nnfit" / usr_path.stem).rmdir()
+
+
+def load_fit(usr_path):
+    """
+    Loads all the replica pdfs at fitting scale in usr_path and return the exportgrids
+
+    Parameters
+    ----------
+
+        usr_path: pathlib.Path
+            path to the folder containing the fit
+
+    Returns
+    -------
+
+            : dict
+            exportgrids info
+    """
+    nnfitpath = usr_path / "nnfit"
+    replica_list = []
+    for yaml_file in nnfitpath.glob("replica_*/*.exportgrid"):
+        replica_list.append(yaml_file)
+    pdf_dict = {}
+    for replica in replica_list:
+        data = yaml.safe_load(replica.read_text())
+        pdf_dict[replica.parent.stem] = data
+    return pdf_dict
+
+
+def evolve_exportgrid(exportgrid, eko, x_grid):
+    """
+    Evolves the provided exportgrid for the desired replica with the eko and returns the evolved block
+
+    Parameters
+    ----------
+        exportgrid: dict
+            exportgrid of pdf at fitting scale
+        eko: eko object
+            eko operator for evolution
+        xgrid: list
+            xgrid to be used as the targetgrid
+    Returns
+    -------
+        : np.array
+        evolved block
+    """
+    # construct LhapdfLike object
+    pdf_grid = np.array(exportgrid["pdfgrid"]).transpose()
+    pdf_to_evolve = utils.LhapdfLike(pdf_grid, exportgrid["q20"], x_grid)
+    # evolve pdf
+    evolved_pdf = apply.apply_pdf(eko, pdf_to_evolve)
+    # generate block to dump
+    targetgrid = list(eko["targetgrid"])
+    def ev_pdf(pid, x, Q2):
+        return x * evolved_pdf[Q2]["pdfs"][pid][targetgrid.index(x)]
+    block = genpdf.generate_block(
+        ev_pdf,
+        xgrid=targetgrid,
+        Q2grid=list(eko["Q2grid"].keys()),
+        pids=br.flavor_basis_pids,
+    )
+    return block
+
+
+def dump_evolved_replica(evolved_block, usr_path, replica_num):
+    """
+    Dump the evolved replica given by evolved_block as the replica num "replica_num" in
+    the folder usr_path/nnfit/usr_path.stem.
+
+    Parameters
+    ----------
+        evolved_block: numpy.array
+            block of an evolved PDF
+        usr_path: pathlib.Path
+            path of the fit folder
+        replica_num: int
+            replica number
+    """
+    path_where_dump = usr_path / "nnfit" / usr_path.stem
+    # create folder to dump the evolved replica if it does not exist
+    path_where_dump.mkdir(exist_ok=True)
+    to_write_in_head = "PdfType: replica\nFromMCReplica: " + str(replica_num) + "\n"
+    genpdf.export.dump_blocks(
+        path_where_dump, replica_num, [evolved_block], pdf_type=to_write_in_head
```

Comment:
I still don't understand why instead of `path_were_dump` you cannot simply tell it `nnfit/replica_{replica_num}`?